documentation/docs/protocol_specification/components/congestion_control.md

```diff
@@ -1,87 +1,173 @@
-
-Here a fundamental remark: _the network manager sets up a desired maximum (fixed) rate_ `SCHEDULING_RATE` _at which messages will be scheduled_, computed in weight (see above) per second. This implies that every message is scheduled after a delay which is equal to the weight (size as default) of the latest scheduled message times the parameter `SCHEDULING_RATE`. This rate mostly depends on the degree of decentralization desired: e.g., a larger rate leads to higher throughput but would leave behind slower devices which will fall out of sync.
+Scheduling is the most critical task in the congestion control component. The scheduling algorithm must guarantee that
+an honest node `node` meets the following requirements:
+
+* __Consistency__: The node's messages will not accumulate indefinitely at any node, and so, starvation is avoided.
+* __Fairness__: The node's fair share of the network resources are allocated to it according to its access Mana.
+* __security__: Malicious nodes sending above their allowed rate will not interrupt a node's throughput requirement.
+
+Although nodes in our setting are capable of more complex and customised behaviour than a typical router in a
+packet-switched network, our scheduler must still be lightweight and scalable due to the potentially large number of
+nodes requiring differentiated treatment. It is estimated that over 10,000 nodes operate on the Bitcoin network, and
+we expect that an even greater number of nodes are likely to be present in the IoT setting. For this reason, we
+adopt a scheduler based on [Deficit Round Robin](https://ieeexplore.ieee.org/document/502236) (DRR) (the Linux
+implementation of the [FQ-CoDel packet scheduler](https://tools.ietf.org/html/rfc8290), which is based on DRR,
+supports anywhere up to 65535 separate queues).
+
+The DRR scans all non-empty queues in sequence. When it selects a non-empty queue, the DDR will increment the queue's
+priority counter (_deficit_) by a specific value (_quantum_). Then, the value of the deficit counter is a maximal amount
+of bytes that can be sent this turn. If the deficit counter is greater than the weight of the message at the head of the
+queue, the DRR can schedule this message, and this weight decrements the value of the counter. In our implementation,
+the quantum is proportional to the node's access Mana, and we add a cap on the maximum deficit that a node can achieve
+to keep the network latency low. It is also important to mention that the DRR can assign the weight of the message so
+that specific messages can be prioritized (low weight) or penalized (large weight); by default, in our mechanism, the
+weight is proportional to the message size measured in bytes. The weight of a message is set by the
+function `WorkCalculator()`.
+
+:::note
```

Comment:
Is this a special markdown directive?

Reply:
Yes. I think it's for the documentation on the website.